ibi-group/datatools-ui
# chore(lint): Add pre-commit hook to enforce formatting.

### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [na] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds a pre-commit hook to run `eslint` on files specified using `git add`. `mastarm` rules are used (see also https://github.com/conveyal/mastarm/pull/310) (this PR uses the existing mastarm dependency).

The PR is very similar to https://github.com/opentripplanner/otp-react-redux/pull/377.

Assumptions:
- You have eslint installed system-wide (otherwise, probably run `yarn global add eslint`)

To test this PR after checkout:
1. Run `yarn install`
2. Add blank spaces/lines to a `.js` file
3. Commit the changes
4. Observe that your extra spaces have been fixed, e.g.
![image](https://user-images.githubusercontent.com/56846598/120685124-82362d00-c46d-11eb-9471-eeca56c33773.png)
5. Run `git reset HEAD~` to undo your commit and undo your file changes.

This PR uses a version of husky that is more recent than in OTP-UI, and the way it is configured is thus different. Specifically, husky introduces:
- hook files and its own config directory
- a `postinstall` script that is ran after install.
